code/modules/unit_tests/cargo_dep_order_locations.dm

```diff
@@ -0,0 +1,20 @@
+/datum/unit_test/cargo_dep_order_locations
+
+/datum/unit_test/cargo_dep_order_locations/Run()
+	for(var/datum/job_department/department as anything in subtypesof(/datum/job_department))
+		var/datum/job_department/dep = new department(FALSE)
+		var/delivery_areas = dep.department_delivery_areas
+		if(isnull(initial(dep.department_delivery_areas)) || !length(delivery_areas))
+			continue
+		if(check_valid_delivery_location(delivery_areas))
+			continue
+		else
+			TEST_FAIL("[department.type] failed to find a valid delivery location on this map.")
```

Comment:
And if we move all the code to this unit test, we don't need to `new` the departments, we can just check `SSjobs`

Reply:
Valid points, on it